main.rb

```diff
@@ -0,0 +1,107 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << i if yield i
+    end
+    arr
+  end
+
+  def my_all?(*args, &block)
+    if block_given?
+      my_select(&block).length == length
+    else
+      return true if args == []
+
+      return (my_select { |i| i.instance_of?(args[0]) }).length == length if args[0].is_a? Class
+      return (my_select { |i| i.match(args[0]) }).length == length if args[0].is_a? Regexp
+      return (my_select { |i| i == args[0] }).length == length if args[0].is_a? Integer
+
+      false
+    end
+  end
```

Comment:
- [ ] it should return false if the block returns false or nil::range

```Ruby
  range.my_all?(&false_block) # returns undefined local variable or method `length' for 5..50:Range
```

- [ ] when no block or argument is given it should return false when one of the collection members are false or nil

```Ruby
  false_array.my_all?

  # expected false
  # got true
```

- [ ] when a class is passed as an argument it should return true if all of the collection is a member of such class::Numeric

```Ruby
  numbers.my_all?(Numeric)

  # expected true
  # got false
```